contracts/managers/SocialTradingManager.sol

```diff
@@ -0,0 +1,296 @@
+     * @param _startingValue                    Starting value of one share of the trading pool to 18 decimals of precision
+     * @param _name                             The name of the new RebalancingSetTokenV2
+     * @param _symbol                           The symbol of the new RebalancingSetTokenV2
+     * @param _rebalancingSetCallData           Byte string containing additional call parameters to pass to factory
+     */
+    function createTradingPool(
+        ISocialAllocator _tradingPairAllocator,
+        address _factory,
+        uint256 _startingBaseAssetAllocation,
+        uint256 _startingValue,
+        bytes32 _name,
+        bytes32 _symbol,
+        bytes calldata _rebalancingSetCallData
+    )
+        external
+    {
+        require(
+            _startingBaseAssetAllocation <= SCALE_FACTOR,
+            "SocialTradingManager.createTradingPool: Starting allocation is not valid."
+        );
+
+        ISetToken collateralSet = _tradingPairAllocator.determineNewAllocation(
+            _startingBaseAssetAllocation,
+            SCALE_FACTOR
+        );
+        
+        uint256[] memory units = new uint256[](1);
+
+        uint256 collateralValue = _tradingPairAllocator.calculateCollateralSetValue(
+            collateralSet
+        );
+
+        units[0] = _startingValue.mul(REBALANCING_SET_NATURAL_UNIT).div(collateralValue);
+
+        address[] memory components = new address[](1);
+        components[0] = address(collateralSet);
+
+        address tradingPool = core.createSet(
+            _factory,
+            components,
+            units,
+            REBALANCING_SET_NATURAL_UNIT,
+            _name,
+            _symbol,
+            _rebalancingSetCallData
+        );
+
+        pools[tradingPool].trader = msg.sender;
+        pools[tradingPool].allocator = _tradingPairAllocator;
+        pools[tradingPool].currentAllocation = _startingBaseAssetAllocation;
+
+        emit TradingPoolCreated(
+            msg.sender,
+            _tradingPairAllocator,
+            tradingPool,
+            _startingBaseAssetAllocation
+        );
+    }
+
+    /*
+     * Update trading pool allocation. Issue new collateral Set and pass on to RebalancingSetTokenV2.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newAllocation      New base asset allocation in a scaled decimal value
+     *                                          (e.g. 100% = 1e18, 1% = 1e16)
+     */
+    function updateAllocation(
+        IRebalancingSetTokenV2 _tradingPool,
+        uint256 _newAllocation
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        validateAllocationUpdate(_tradingPool, _newAllocation);
+
+        ISetToken nextSet = pools[address(_tradingPool)].allocator.determineNewAllocation(
+            _newAllocation,
+            SCALE_FACTOR
+        );
+
+        _tradingPool.startRebalance(address(nextSet));
```

Comment:
As you know: With the new changes, it has the `liquidatorData` argument.

Reply:
+1